pulp_2to3_migrate/app/json_schema.py

```diff
@@ -0,0 +1,89 @@
+SCHEMA='''{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "MigrationPlan",
+    "type": "object",
+    "properties": {
+        "plugins": {
+            "type": "array",
+            "minItems": 1,
+            "items": {
+                "type": "object",
+                "properties": {
+                    "type": {
+                        "type": "string"
+                    },
+                    "protection": {
+                        "type": "boolean"
+                    },
+                    "repositories": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "type": "string"
+                                },
+                                "pulp2_repository_id": {
+                                    "type": "string"
+                                },
+                                "repository_versions": {
+                                    "type": "array",
+                                    "minItems": 1,
+                                    "items": {
+                                        "type": "object",
+                                        "properties": {
+                                            "pulp2_repository_id": {
+                                                "type": "string"
+                                             },
+                                             "distributor_ids": {
+                                                 "type": "array"
+                                             },
+                                             "protection": {
+                                                 "type": "array"
+                                              }
+                                        },
+                                        "required": ["pulp2_repository_id"],
+                                        "additionalProperties": false,
+                                        "dependencies": {
+                                            "protection": [ "distributor_ids"]
+                                        }
+                                    }
+                                }
+                            },
+                            "required": ["name", "repository_versions"],
+                            "additionalProperties": false,
+			    "if": {
```

Comment:
That's a cool validation.
Do you know if there are comments as a part of the schema syntax?
It would be really nice to leave comments what the requirement of certain condition is for.
If not, than maybe at the beginning of the file in a docstring, we can list what we check and why - at least for non-onbious things like this one. What do you think?

Reply:
i added a $comment field